src/views/News/index.js

```diff
@@ -19,530 +23,764 @@ import {
+  const [newPostSubject, setNewPostSubject] = useState('');
+  const [newPostBody, setNewPostBody] = useState('');
+  const [showCropper, setShowCropper] = useState(null); //null if no picture chosen, else contains picture
+  const [photoDialogErrorTimeout, setPhotoDialogErrorTimeout] = useState(null);
+  const [photoDialogError, setPhotoDialogError] = useState(null);
+  const [cropBoxDim, setCropBoxDim] = useState(null);
+  const [aspectRatio, setAspectRatio] = useState(null);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('Something went wrong');
+  const [currentUsername, setCurrentUsername] = useState('');
+  const [currentlyEditing, setCurrentlyEditing] = useState(false); // false if not editing, newsID if editing
+  const [justShowPicture, setJustShowPicture] = useState(false);
+  /*justShowPicture is a more complicated state property. If an image was submitted
+   *to a news post, then if the user clicks on the edit button for the post later,
+   *we want to at first just show them the image, not the cropper (particularly since
+   *the cropper would only be able to contain what the API stored, so they can only
+   *crop more). If they remove the picture, then we want to show the dropzone. If they
+   *add a new picture, we want to display the cropper with the new data. When edit is
+   *clicked for the first time and when a new picture has been added, showCropper will
+   *have data, so we can't use showCropper to determine whether or not to show an img
+   *tag or a cropper. A new property was necessary. justShowPicture begins false and is
+   *made true when the edit button is clicked. When either the remove picture or cancel
+   *button is clicked from the edit window, it is set back to false. In this way there is
+   *the ability to update the view to the user. As a side note, this also creates the
+   *ability to change the message above the image in a way that makes sense in either the
+   *just-opened-an-edit or the just-submitted-new-image context.
+   */
+  let cropperRef = React.createRef();
+
+  const styles = {
+    button: {
+      background: gordonColors.primary.blue,
+      color: 'white',
+
+      changeImageButton: {
+        background: gordonColors.primary.blue,
+        color: 'white',
+      },
+
+      resetButton: {
+        backgroundColor: '#f44336',
+        color: 'white',
+      },
+      cancelButton: {
+        backgroundColor: 'white',
+        color: gordonColors.primary.blue,
+        border: `1px solid ${gordonColors.primary.blue}`,
+        width: showCropper ? '38%' : '86%',
+      },
+      hidden: {
+        display: 'none',
+      },
+    },
+    searchBar: {
+      margin: '0 auto',
+    },
+    newNewsForm: {
+      backgroundColor: '#fff',
+    },
+    fab: {
+      margin: 0,
+      top: 'auto',
+      right: 40,
+      bottom: 40,
+      left: 'auto',
+      position: 'fixed',
+      zIndex: 1,
+    },
+  };
```

Comment:
I think the best way to do this is to have styles defined outside the component, and then here you can set `styles.button.cancelButton.width = showCropper ? '38%' : '86%'`.